src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let res = this[0];
+
+    if (!this.length) {
+      return '';
+    }
+
+    if (res === null) {
+      res = '';
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        res += separator + '';
+        continue;
+      }
+      res += String(separator) + this[i];
```

Comment:
Also you should separator if it's not the last element
```suggestion
      if (this[i] !== null || this[i] !== undefined) {
        res += this[i];
      }
      
      if (i < this.length - 1) {
        result += separator;
      }
     
```